swift/probe/Sources/moonlight.swift

```diff
@@ -79,16 +78,28 @@ struct System {
         task.waitUntilExit()
         return task.terminationStatus
     }
-    func executable(url: URL) {
+    func executable(url: URL) -> Optional<URL> {
+        var path = URL(fileURLWithPath: url.path)
+        #if os(Windows)
+        let range = url.lastPathComponent.range(of: ".tmp")!
+        path = URL(string: "\(url.lastPathComponent[..<range.lowerBound]).exe", relativeTo: path)!
```

Comment:
since you already have this PR, I'll just add here instead of opening my own :)
I came up with the same basic answer, but if it helps:

can replace the .tmp extension with: 
```
path = url.deletingPathExtension().appendingPathExtension("exe")
```

and a nit, but can use `FileManager.default.moveItem` instead of `copyItem`

Reply:
moveItem will fail if the file already exists on disk, which happened during testing